### DISCUSSION TITLE
New version 0.14. CW functionality.

### DISCUSSION THREAD

**Discussion by dl2ki** [foldynl/QLog#119] [General]
The CW functionality in the new version 0.14 works very well. CW via the CAT interface and CW via my K3NG keyer with WinKey2 emulation as well.

A few small things caught my eye, but I need to take a closer look. Some new things also take some getting used to. 

Congratulations for the excellent work on the software

**Comment by foldynl** [comment]
Thank you very much for your comments. I am also using K3NG in emulation mode for testing the QLog. 

I look forward to your comments. To be honest, I had no idea what all had to be done - the change was quite big

**Comment by mppks** [comment]
Hi, please add to CW Keyers the ability to control CW keying using serial port DTR/RTS pin, for example, like in Fldigi or CwType.
http://www.w1hkj.com/FldigiHelp/cw_dtr_rts_keying.html
![Fldigi](https://github.com/foldynl/QLog/assets/3379287/07398187-c006-4c1c-85f7-5b151ed391c6)
![CWType](https://github.com/foldynl/QLog/assets/3379287/baf05c80-d99e-4cf4-a489-0e8f51de9882)

**Reply by foldynl** [reply]
fldigi or cwtype are not LOG applications. Likewise, QLog is not a Keyer application. Therefore, I do not currently plan to implement CW at a low level (direct Morse code generator in QLog). QLog supports Hamlib or WinKey interface. Possible cwdaemon support can be implemented (I don't know if there is a version for Windows). 

However, any Pull Request related to this area will be welcomed.

**Reply by mppks** [reply]
Ok

**Comment by sq5jup** [comment]
Hi,
Is it possible to add support for WinKey for version 1?

73 de SQ5J,
Piotr

**Comment by foldynl** [comment]
I don't plan to add this in the near feature, because the TODO list is very long, but I will look at the Winkey1 specification what it would mean to add winkey1 support. If there are no fundamental differences between winkey2, and I will be able to simulate it on K3NG, then it's possible to implement it. 

But honestly, wouldn't it be better to use a more modern [K3NG keyer](https://blog.radioartisan.com/arduino-cw-keyer/) with Winkey2 support than a Winkey1 chip that's 20 years old?

**Reply by sq5jup** [reply]
Thank you for the quick response. The argument for adding version 1 is that the company [microHAM ](https://www.microham.com/) has probably sold around 100K or more microKEYER II devices, which support version 1. Many CW operators may not want to invest in new equipment. Of course, I don't want to put any pressure; it's just my request. I will, of course, follow the changes in the program, as it seems to be very good.
Thank you in advance.
